src/Task/Logfile/RotateLog.php

```diff
@@ -40,12 +39,12 @@ public function __construct($logfiles)
     /**
      * @param int $keep
      * @return RotateLog
-     * @throws InvalidArgumentException
+     * @throws \Exception
      */
     public function keep(int $keep): self
     {
         if ($keep < 1) {
-            throw new InvalidArgumentException(
```

Comment:
The side effect was that the `robo docs` command generates sub folders in the auto generated docs for the sub folders in the task dir `Task\Logfile`. In my opinion we can hold the custom exception, it is more clean and we can better handle the exception.

What do you think?

Reply:
More specific exceptions are preferable to more generic ones. I am generally not in favor of creating custom exceptions unless there is a specific use-case where the client is going to want to catch it explicitly, but in cases where there is an existing exception that is suitable, it is helpful to use it.

Perhaps the docs tool was generating extraneous subfolders because of a typo, `InvalidArgumentException` instead of `\ InvalidArgumentException`.

Reply:
You're right. I've decided to change the global `Exception` to a global `InvalidArgumentException`. A custom exception isn't really needed here. But feel free to change this in future, for me both are OK.